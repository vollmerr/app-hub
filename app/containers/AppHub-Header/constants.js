/*
 *
 * AppHubHeader constants
 *
 */

export const DEFAULT_ACTION = 'app/AppHubHeader/DEFAULT_ACTION';
