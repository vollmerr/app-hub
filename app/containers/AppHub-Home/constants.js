/*
 *
 * AppHubHome constants
 *
 */

export const DEFAULT_ACTION = 'app/AppHubHome/DEFAULT_ACTION';
