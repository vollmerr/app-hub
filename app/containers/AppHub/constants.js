// VIEW
export const CHANGE_MOBILE = 'AppHub/CHANGE_MOBILE';
export const CHANGE_PANEL_OPEN = 'AppHub/CHANGE_PANEL_OPEN';
export const CHANGE_PANEL_SELECTED = 'AppHub/CHANGE_PANEL_SELECTED';
// panel names
export const HELP_PANEL = 'AppHub/HELP_PANEL';
export const APPS_PANEL = 'AppHub/APPS_PANEL';
export const APP_NAV_PANEL = 'AppHub/APP_NAV_PANEL';
export const ALERTS_PANEL = 'AppHub/ALERTS_PANEL';
export const DEV_PANEL = 'AppHub/DEV_PANEL';
// APP
export const CHANGE_APP = 'AppHub/CHANGE_APP';
export const CHANGE_APP_STATUS = 'AppHub/CHANGE_APP_STATUS';
// user
export const AUTH_USER = 'AppHub/AUTH_USER';
export const AUTH_USER_DONE = 'AppHub/AUTH_USER_DONE';
// REQUEST STATUS TYPES
export const REQUEST = '_REQUEST';
export const SUCCESS = '_SUCCESS';
export const FAILURE = '_FAILURE';

export const COL_TYPES = {
  DATE: 'DATE',
};

export const ROLES = {
  USER: 'AppHub User',
};

// D3
export const INCREMENT_RENDER_COUNT = 'AppHub/INCREMENT_RENDER_COUNT';
