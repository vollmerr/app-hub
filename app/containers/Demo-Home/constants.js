export const EXAMPLE_DATA_REQUEST = 'app/DemoHome/EXAMPLE_DATA_REQUEST';
export const EXAMPLE_DATA_SUCCESS = 'app/DemoHome/EXAMPLE_DATA_SUCCESS';
export const EXAMPLE_DATA_FAILURE = 'app/DemoHome/EXAMPLE_DATA_FAILURE';
