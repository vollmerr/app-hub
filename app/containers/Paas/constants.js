export const GET_MANAGER_DATA_REQUEST = 'app/Paas/GET_MANAGER_DATA_REQUEST';
export const GET_MANAGER_DATA_SUCCESS = 'app/Paas/GET_MANAGER_DATA_SUCCESS';
export const GET_MANAGER_DATA_FAILURE = 'app/Paas/GET_MANAGER_DATA_FAILURE';

export const UPDATE_USERS_REQUEST = 'app/Paas/UPDATE_USERS_REQUEST';
export const UPDATE_USERS_SUCCESS = 'app/Paas/UPDATE_USERS_SUCCESS';
export const UPDATE_USERS_FAILURE = 'app/Paas/UPDATE_USERS_FAILURE';

export const GET_REPORT_DATA_REQUEST = 'app/Paas/GET_REPORT_DATA_REQUEST';
export const GET_REPORT_DATA_SUCCESS = 'app/Paas/GET_REPORT_DATA_SUCCESS';
export const GET_REPORT_DATA_FAILURE = 'app/Paas/GET_REPORT_DATA_FAILURE';

export const SET_REPORT_KEY = 'app/Paas/SET_REPORT_KEY';
export const SET_REPORT_FILTER = 'app/Paas/SET_REPORT_FILTER';

// ROLES
export const ROLES = {
  MANAGER: 'PAAS Manager',
  SECURITY: 'PAAS Security',
  HR: 'PAAS HR',
};
export const ROLES_VALUES = Object.values(ROLES);
// APPROVAL CODES
export const APPROVAL = {
  NONE: undefined,
  DENY: 0,
  APPROVE: 1,
};
// ACCESS AUTHORIZATION
export const AUTH = {
  ID: '_id',
  SID: 'sid',
  FULL_NAME: 'fullName',
  EMAIL: 'email',
  POS_NUMBER: 'posNumber',
  MANAGER_SID: 'managerSid',
  MANAGER_NAME: 'managerFullName',
  STATUS: 'status',
  LAST_MODIFIED: 'lastUpdated',
  LAST_APPROVED: 'lastApproved',
  DATE_CREATED: 'created',
  AUTH_YEAR: 'authYear',
};
// AUTH STATUS
export const STATUS = {
  ACTIVE: 'active',
  INACTIVE: 'inactive',
  NO_MANAGER: 'noManager',
};
// APP TO AUTHORIZE
export const APPS = {
  APP_1: 'app1',
  APP_2: 'app2',
  APP_3: 'app3',
  APP_4: 'app4',
};
export const APP_LIST = Object.values(APPS);
export const BUTTONS = {
  APPROVE: 'approveButton',
  DENY: 'denyButton',
};
export const REPORT = {
  APPROVED: 0,
  DENIED: 1,
  PENDING: 2,
  NO_MANAGER: 3,
};
