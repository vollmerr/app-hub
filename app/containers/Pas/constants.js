export const GET_MANAGER_DATA_REQUEST = 'app/Pas/GET_MANAGER_DATA_REQUEST';
export const GET_MANAGER_DATA_SUCCESS = 'app/Pas/GET_MANAGER_DATA_SUCCESS';
export const GET_MANAGER_DATA_FAILURE = 'app/Pas/GET_MANAGER_DATA_FAILURE';

export const UPDATE_USERS_REQUEST = 'app/Pas/UPDATE_USERS_REQUEST';
export const UPDATE_USERS_SUCCESS = 'app/Pas/UPDATE_USERS_SUCCESS';
export const UPDATE_USERS_FAILURE = 'app/Pas/UPDATE_USERS_FAILURE';

export const GET_REPORT_DATA_REQUEST = 'app/Pas/GET_REPORT_DATA_REQUEST';
export const GET_REPORT_DATA_SUCCESS = 'app/Pas/GET_REPORT_DATA_SUCCESS';
export const GET_REPORT_DATA_FAILURE = 'app/Pas/GET_REPORT_DATA_FAILURE';

export const SET_REPORT_KEY = 'app/Pas/SET_REPORT_KEY';
export const SET_REPORT_FILTER = 'app/Pas/SET_REPORT_FILTER';

export const UPDATE_USER_MANAGER_REQUEST = 'app/Pas/UPDATE_USER_MANAGER_REQUEST';
export const UPDATE_USER_MANAGER_SUCCESS = 'app/Pas/UPDATE_USER_MANAGER_SUCCESS';
export const UPDATE_USER_MANAGER_FAILURE = 'app/Pas/UPDATE_USER_MANAGER_FAILURE';


// ROLES
export const ROLES = {
  MANAGER: 'PAAS Manager',
  SECURITY: 'PAAS Security',
  HR: 'PAAS HR',
};
export const ROLES_VALUES = Object.values(ROLES);
// APPROVAL CODES
export const APPROVAL = {
  NONE: undefined,
  DENY: 0,
  APPROVE: 1,
};
// ACCESS AUTHORIZATION
export const AUTH = {
  ID: '_id',
  SID: 'sid',
  FULL_NAME: 'fullName',
  EMAIL: 'email',
  POS_NUMBER: 'posNumber',
  MANAGER_SID: 'managerSID',
  MANAGER_NAME: 'managerFullName',
  STATUS: 'status',
  LAST_MODIFIED: 'lastUpdated',
  LAST_APPROVED: 'lastApproved',
  DATE_CREATED: 'created',
  AUTH_YEAR: 'authYear',
};
// AUTH STATUS
export const STATUS = {
  ACTIVE: 'active',
  INACTIVE: 'inactive',
  NO_MANAGER: 'noManager',
  ASSIGNED_MANAGER: 'assignedManager',
};
// APP TO AUTHORIZE
export const APPS = {
  APP_1: 'app1',
  APP_2: 'app2',
  APP_3: 'app3',
  APP_4: 'app4',
};
export const APP_LIST = Object.values(APPS);
export const BUTTONS = {
  APPROVE: 'approveButton',
  DENY: 'denyButton',
};
export const REPORT = {
  APPROVED: 0,
  DENIED: 1,
  PENDING: 2,
  NO_MANAGER: 3,
};

export const MANAGE = {
  EMPLOYEE_ID: 'employee_id',
  MANAGER_ID: 'manager_id',
};
