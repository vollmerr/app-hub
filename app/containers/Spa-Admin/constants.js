export const FIELD_ACK_NAME = 'name';
export const FIELD_ACK_FILE = 'file';
