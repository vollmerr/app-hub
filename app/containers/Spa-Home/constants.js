/*
 *
 * SpaHome constants
 *
 */

export const DEFAULT_ACTION = 'app/SpaHome/DEFAULT_ACTION';
