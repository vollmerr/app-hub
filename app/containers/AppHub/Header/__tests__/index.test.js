import React from 'react';
import { shallow } from 'enzyme';

import { testStyledComponent } from '../../../../utils/testUtils';

import * as C from '../../constants';

import { Header, Wrapper, Section, Line } from '../index';
import UserInfo from '../UserInfo';
import Logo from '../Logo';


testStyledComponent(Wrapper);
testStyledComponent(Section);
testStyledComponent(Line);


describe('<Line/>', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Line />);
  });

  it('should render correctly if is `partial`', () => {
    wrapper.setProps({ partial: true });
    expect(wrapper).toMatchSnapshot();
  });
});


global.isDev = false;

const props = {
  app: {
    name: 'app name',
    homePath: '/test',
  },
  user: {
    name: 'user name',
  },
  view: {
    isMobile: false,
  },
  panel: {
    name: 'test panel',
    isOpen: false,
  },
  onClick: jest.fn(),
};

const testIfChecked = (wrapper, name) => {
  wrapper.setProps({ panel: { name, isOpen: true } });
  expect(wrapper.find({ panel: name }).prop('checked')).toEqual(true);
};


describe('<Header />', () => {
  let wrapper;
  let instance;
  beforeEach(() => {
    wrapper = shallow(
      <Header {...props} />
    );
    instance = wrapper.instance();
  });

  describe.only('renderLink', () => {
    let linkWrapper;
    it.only('should render a checked `NavLink` if the selected panel is open', () => {
      const linkProps = {
        panel: props.panel.name,
      };
      wrapper.setProps({ panel: { ...props.panel, isOpen: true } });
      linkWrapper = shallow(instance.renderLink(linkProps));
      expect(linkWrapper.prop('checked')).toEqual(true);
    });

    it('should render an unchecked `NavLink` if is not open or not the panel selected', () => {
      let linkProps = {
        panel: 'other panel',
      };
      wrapper.setProps({ panel: { ...props.panel, isOpen: true } });
      linkWrapper = shallow(instance.renderLink(linkProps));
      expect(linkWrapper.prop('checked')).toEqual(false);

      linkProps = {
        panel: props.panel.name,
      };
      linkWrapper = shallow(instance.renderLink(linkProps));
      expect(linkWrapper.prop('checked')).toEqual(false);
    });
  });

  describe('render - desktop view', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render two sections (left and right)', () => {
      expect(wrapper.find(Section).length).toEqual(2);
    });

    it('should not render the app nav hamburger link', () => {
      const panel = C.APP_NAV_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(0);
    });

    it('should render the logo', () => {
      expect(wrapper.find(Logo).length).toEqual(1);
    });

    it('should render the App Hub title/icon', () => {
      const text = 'App Hub';
      expect(wrapper.find({ text }).length).toEqual(1);
    });

    it('should render the app title if passed an `app.name`', () => {
      const text = props.app.name;
      expect(wrapper.find({ text }).length).toEqual(1);

      wrapper.setProps({ app: {} });
      expect(wrapper.find({ text }).length).toEqual(0);
    });

    it('should render the user information if passed a `user.name`', () => {
      expect(wrapper.find(UserInfo).length).toEqual(1);

      wrapper.setProps({ user: {} });
      expect(wrapper.find(UserInfo).length).toEqual(0);
    });

    it('should generate the correct user intitials', () => {
      const initials = 'UN'; // for 'User Name'
      expect(wrapper.find(UserInfo).prop('initials')).toEqual(initials);
    });

    it('should render the dev panel link if in development mode', () => {
      try {
        const panel = C.DEV_PANEL;
        expect(wrapper.find({ panel }).length).toEqual(0);

        global.isDev = true;
        wrapper = shallow(<Header {...props} />);

        expect(wrapper.find({ panel }).length).toEqual(1);
        testIfChecked(wrapper, panel);
      } finally {
        global.isDev = false; // reset to not dev mode for other tests.
      }
    });

    it('should render the dev panel link if in mock mode', () => {
      try {
        const panel = C.DEV_PANEL;
        expect(wrapper.find({ panel }).length).toEqual(0);

        global.isMock = true;
        wrapper = shallow(<Header {...props} />);

        expect(wrapper.find({ panel }).length).toEqual(1);
        testIfChecked(wrapper, panel);
      } finally {
        global.isMock = false; // reset to not dev mode for other tests.
      }
    });

    // it('should render the alerts panel link if passed `alerts`', () => {
    //   const panel = ALERTS_PANEL;
    //   expect(wrapper.find({ panel }).length).toEqual(1);

    //   testIfChecked(wrapper, panel);

    //   wrapper.setProps({ alerts: [] });
    //   expect(wrapper.find({ panel }).length).toEqual(0);
    // });

    it('should render the help panel link', () => {
      const panel = C.HELP_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(1);
      testIfChecked(wrapper, panel);
    });

    it('should render the App Hub navigation link', () => {
      const panel = C.APPS_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(1);
      testIfChecked(wrapper, panel);
    });
  });

  describe('render - mobile view', () => {
    beforeEach(() => {
      const mobileProps = { ...props, view: { isMobile: true } };
      wrapper = shallow(
        <Header {...mobileProps} />
      );
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render two sections (left and right)', () => {
      expect(wrapper.find(Section).length).toEqual(2);
    });

    it('should render app nav link if passed a `app.name`', () => {
      const panel = C.APP_NAV_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(1);
      testIfChecked(wrapper, panel);

      wrapper.setProps({ app: {} });
      expect(wrapper.find({ panel }).length).toEqual(0);
    });

    it('should not render the logo', () => {
      expect(wrapper.find(Logo).length).toEqual(0);
    });

    it('should not render the App Hub title/icon', () => {
      const text = 'App Hub';
      expect(wrapper.find({ text }).length).toEqual(0);
    });

    it('should render the app title if passed one', () => {
      const text = props.app.name;
      expect(wrapper.find({ text }).length).toEqual(1);

      wrapper.setProps({ app: {} });
      expect(wrapper.find({ text }).length).toEqual(0);
    });

    it('should not render the user information', () => {
      expect(wrapper.find(UserInfo).length).toEqual(0);
    });

    it('should render the dev panel link if in development mode', () => {
      try {
        const panel = C.DEV_PANEL;
        expect(wrapper.find({ panel }).length).toEqual(0);

        global.isDev = true;
        wrapper = shallow(<Header {...props} />);
        testIfChecked(wrapper, panel);
        expect(wrapper.find({ panel }).length).toEqual(1);
      } finally {
        global.isDev = false; // reset to not dev mode for other tests.
      }
    });

    //   it('should render the alerts panel link if passed `alerts`', () => {
    //     const panel = ALERTS_PANEL;
    //     expect(wrapper.find({ panel }).length).toEqual(1);

    //     testIfChecked(wrapper, panel);

    //     wrapper.setProps({ alerts: [] });
    //     expect(wrapper.find({ panel }).length).toEqual(0);
    //   });

    it('should render the help panel link', () => {
      const panel = C.HELP_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(1);
      testIfChecked(wrapper, panel);
    });

    it('should render the App Hub navigation link', () => {
      const panel = C.APPS_PANEL;
      expect(wrapper.find({ panel }).length).toEqual(1);
      testIfChecked(wrapper, panel);
    });
  });
});
