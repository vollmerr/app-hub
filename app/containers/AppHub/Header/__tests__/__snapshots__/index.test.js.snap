// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> desktop view should render correctly 1`] = `
<Header__Wrapper>
  <Header__Section>
    <Logo />
    <Header__Line
      partial={true}
    />
    <NavLink
      checked={false}
      iconProps={
        Object {
          "iconName": "Apphub",
          "style": Object {
            "fontSize": "24px",
            "marginRight": "10px",
          },
        }
      }
      onClick={[Function]}
      padding="15px"
      text="App Hub"
      title="App Hub"
      to="/"
    />
    <Header__Line
      partial={true}
    />
    <NavLink
      checked={false}
      onClick={[Function]}
      padding="15px"
      text="app name"
      title="app name"
      to="/test"
    />
  </Header__Section>
  <Header__Section>
    <UserInfo
      initials="UN"
      name="user name"
    />
    <Header__Line />
    <NavLink
      checked={false}
      iconProps={
        Object {
          "iconName": "helpMenu",
        }
      }
      onClick={[Function]}
      panel="AppHub/HELP_PANEL"
      title="Help Panel"
    />
    <Header__Line />
    <NavLink
      checked={false}
      dark={true}
      iconProps={
        Object {
          "iconName": "appHubMenu",
        }
      }
      onClick={[Function]}
      panel="AppHub/APPS_PANEL"
      title="App Hub Navigation"
    />
  </Header__Section>
</Header__Wrapper>
`;

exports[`<Header /> mobile view should render correctly 1`] = `
<Header__Wrapper>
  <Header__Section>
    <NavLink
      checked={false}
      iconProps={
        Object {
          "iconName": "appMenu",
        }
      }
      onClick={[Function]}
      panel="AppHub/APP_NAV_PANEL"
      title="app name Navigation"
    />
    <Header__Line
      partial={false}
    />
    <NavLink
      checked={false}
      onClick={[Function]}
      padding="15px"
      text="app name"
      title="app name"
      to="/test"
    />
  </Header__Section>
  <Header__Section>
    <Header__Line />
    <NavLink
      checked={false}
      iconProps={
        Object {
          "iconName": "helpMenu",
        }
      }
      onClick={[Function]}
      panel="AppHub/HELP_PANEL"
      title="Help Panel"
    />
    <Header__Line />
    <NavLink
      checked={false}
      dark={true}
      iconProps={
        Object {
          "iconName": "appHubMenu",
        }
      }
      onClick={[Function]}
      panel="AppHub/APPS_PANEL"
      title="App Hub Navigation"
    />
  </Header__Section>
</Header__Wrapper>
`;

exports[`<Header /> render - desktop view should render correctly 1`] = `
<Header__Wrapper>
  <Header__Section>
    <Logo />
    <Header__Line
      partial={true}
    />
    <NavLink
      checked={false}
      iconProps={
        Object {
          "iconName": "Apphub",
          "style": Object {
            "fontSize": "24px",
            "marginRight": "10px",
          },
        }
      }
      onClick={[Function]}
      padding="15px"
      text="App Hub"
      title="App Hub"
      to="/"
    />
    <Header__Line
      partial={true}
    />
    <NavLink
      checked={false}
      onClick={[Function]}
      padding="15px"
      text="app name"
      title="app name"
      to="/test"
    />
  </Header__Section>
  <Header__Section>
    <UserInfo
      initials="UN"
      name="user name"
    />
    <Header__Line />
    <NavLink
      checked={false}
      iconProps={
        Object {
          "iconName": "helpMenu",
        }
      }
      onClick={[Function]}
      panel="AppHub/HELP_PANEL"
      title="Help Panel"
    />
    <Header__Line />
    <NavLink
      checked={false}
      dark={true}
      iconProps={
        Object {
          "iconName": "appHubMenu",
        }
      }
      onClick={[Function]}
      panel="AppHub/APPS_PANEL"
      title="App Hub Navigation"
    />
  </Header__Section>
</Header__Wrapper>
`;

exports[`<Header /> render - mobile view should render correctly 1`] = `
<Header__Wrapper>
  <Header__Section>
    <NavLink
      checked={false}
      iconProps={
        Object {
          "iconName": "appMenu",
        }
      }
      onClick={[Function]}
      panel="AppHub/APP_NAV_PANEL"
      title="app name Navigation"
    />
    <Header__Line
      partial={false}
    />
    <NavLink
      checked={false}
      onClick={[Function]}
      padding="15px"
      text="app name"
      title="app name"
      to="/test"
    />
  </Header__Section>
  <Header__Section>
    <Header__Line />
    <NavLink
      checked={false}
      iconProps={
        Object {
          "iconName": "helpMenu",
        }
      }
      onClick={[Function]}
      panel="AppHub/HELP_PANEL"
      title="Help Panel"
    />
    <Header__Line />
    <NavLink
      checked={false}
      dark={true}
      iconProps={
        Object {
          "iconName": "appHubMenu",
        }
      }
      onClick={[Function]}
      panel="AppHub/APPS_PANEL"
      title="App Hub Navigation"
    />
  </Header__Section>
</Header__Wrapper>
`;

exports[`<Header__Line /> should render correctly 1`] = `
.c0 {
  width: 1px;
  margin: auto;
  background: #666666;
  height: calc(50px - 2px);
}

<div
  className="c0"
/>
`;

exports[`<Header__Section /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
/>
`;

exports[`<Header__Wrapper /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
  height: 50px;
  box-sizing: border-box;
  background: #212121;
}

<div
  className="c0"
/>
`;

exports[`<Line/> should render correctly if is \`partial\` 1`] = `
.c0 {
  width: 1px;
  margin: auto;
  background: #666666;
  height: calc(50px - 15px);
}

<div
  className="c0"
/>
`;
